test/models/report_test.rb

```diff
@@ -3,7 +3,60 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  setup do
+    @user = users(:alice)
+    @report = reports(:alice_report)
+    @mentioned_report = reports(:bob_report)
+  end
+  test 'editable?は@userが日報作成者ならtrueを返す' do
+    assert @report.editable?(@user)
+  end
+
+  test 'editable?は@userが日報作成者じゃないならfalseを返す' do
+    incorrect_user = users(:bob)
+    assert_not @report.editable?(incorrect_user)
+  end
+
+  test 'created_onはDateクラスを返す' do
+    @report.created_at = Time.zone.now
+    assert_instance_of Date, @report.created_on
+  end
+
+  test '日報の保存時は既存の言及を全削除し、新規の言及を作成する' do
+    assert_equal 1, @report.active_mentions.size
+    assert @report.active_mentions.find_by(mentioned_by_id: @mentioned_report.id)
+
+    new_mentioned_report = reports(:carol_report)
+    @report.content =
+      "http://localhost:3000/reports/#{new_mentioned_report.id}を見ました。"
+    @report.save
```

Comment:
```suggestion
    @report.save!
```

のようにして万一保存に失敗したらエラーが出てすぐ気づけるようにした方がベターです

Reply:
`false`で瑕疵がある状態でプログラムが進んでいくのを防ぐのですね
修正しました！